### ISSUE TITLE
Trouble with VersionPress write permissions

### ISSUE THREAD

**Issue by philvallender** [versionpress/support#14]
Hi there, 

I'm really not that knowledgable about linux or git, but I really want to use VersionPress to prevent bricking the new site I'm building - it's happened several times already. 

I've deployed wordpress on a aws ec2 instance using bitnami. I was able to install git via ssh and this checks out on the versionpress status screen. Now I'm trying to give VP the write permissions it needs but I'm having no joy. 

I'm unable to identify which user should have the rights or which group to add them to. I've done chmods to give u and g write permission on the entire /wordpress dir. And I've done usermod to add root and daemon to the bitnami group, which I thought would do it.

If look at wordpress/tmp now, as user bitnami, i see that the permissions are rxwrxwr-x bitnami daemon

Since the bitnami distro is likely to be quite common, I hoped you could help me.

Thanks, Phil.

**Comment by borekb** [comment]
Hi Phil, we do not have a direct experience with Bitnami yet, could you please contact their support to see what the issue is? Thank you.

**Comment by philvallender** [comment]
Hi Borek, thanks for your reply. Hope you are having a nice weekend!

I certainly will contact bitnami and see if they can help. That said, if I had something to go on, I might be able to find the solution myself. Are you able to share what giving VP permissions in a 'common' ubuntu build is like? Is it a chmod command on a specific folder, for a specific user? Does the command need -r? Anything like that would help. Thanks, Phil.

**Comment by borekb** [comment]
It largely depends on the web server / PHP setup. Sometimes, the web server user will be the one manipulating the files (e.g., when Apache's `mod_php` is used), sometimes, it's another user (like in many php-fpm setups). Generally, you need to find out which user is manipulating the files and then give him the write permissions. The *Permissions* section of our [System Info page](http://docs.versionpress.net/en/troubleshooting/system-info-page) might help.

**Comment by philvallender** [comment]
If I do a ps -ef I get multiple results including root php-fpm master process and daemon php-fpm pool wordpress. Does that point to the right user to give the permissions too?

**Comment by stibi** [comment]
Hello Phil,
what's the stack you are using to run your wordpress instance? I guess it's nginx with php-fpm? As you mentioned php-fpm in your previous comment...

The linux distribution is based on what? Something ubuntu like? I'm asking because in case of centos or redhat, there might (should) be SElinux enabled and that's another layer of possible troubles...

**Comment by philvallender** [comment]
Hi Stibi, 

I have to admit, again, I'm no expert. All I know is:

- Bitnami wordpress cloud image
- Ubuntu distro
- php-fpm is reported in the ps -ef results but I don't know more than this

I haven't come across anything that confirms apache or nginx for me yet.

FYI: https://aws.amazon.com/marketplace/pp/B007IP8BKQ

**Comment by borekb** [comment]
BTW you've probably tried to contact their support too, have they given any clues yet?

**Comment by philvallender** [comment]
No, no reply yet.

**Comment by philvallender** [comment]
Hey guys, 
Got it working. Used sudo chmod -R g+w on two folders, htdocs and tmp and ticks for write access in the VP dashboard. Now I need to sort out the access rules...

**Comment by stibi** [comment]
Hi Phil,
that's great, I'll take a look on the Bitnami image, I don't know how it works, never used it before.

FYI, if you need to find out what is actually running on you virtual machine, try this:

- `ps aux | grep httpd` this returns any apache process running
- `ps aux | grep nginx` this returns any nginx process running
- `ps aux | grep php-fpm` this returns any php-fpm process running

Also look for the user and group under which the processes runs, it will give us more clue how the things are configured. Of course we can look into the configuration files, but I need to find their locations, the php-fpm should be somewhere in `/etc/php-fpm.d/`, or `/etc/php5/fpm/pool.d/` and/or `/etc/php-fpm.conf`'.

For the apache webserver, look into `/etc/apache2`, the `conf.d/`, `sites-enabled/` and `mods-enabled/` should contain something interesting...

By the doc it looks there is the nginx & php-fpm combination:
https://wiki.bitnami.com/Applications/BitNami_Wordpress

As for the access rules, it depends on the web server...

**Comment by philvallender** [comment]
Thanks Stibi, I've learned a lot today!

I'm looking into these access rules. The WP htaccess editor plugin things I have no .htaccess file. If I create one, would a couple of rewrite rules here address the access rule requirements for VP?

Cheers, Phil.

**Comment by borekb** [comment]
Thanks Phil, good luck with your setup and let us know if you get it working - or even submit a PR to our [docs repo](https://github.com/versionpress/docs) detailing the Bitnami setup :)

**Comment by meadow19** [comment]
Hello,

I have a similar problem.

I am using a standard Ubuntu16.04 image with ServerPilot.io on top, with Letsencrypt.org certificates. Then I installed the latest version of Wordpress. 

That's it.

Now, I tried to install Versionpress, but Wordpress didn't let me upload the plugin. I don't have FTP or FTPS installed on the server it seems. So I had to copy manually.

Now, the Versionpress 'check procedure' was all fine, all ticked. But when I installed it, for a second or two it showed a few lines about chmod errors. I didn't get time to read them, because after a second, the page refreshed and the standard Versionpress page showed.

Now I'm worried that Versionpress is not working correctly. How do I check?  There doesn't seem to be a way to go back and check the error messages. Thank you kindly for your help.
